### ISSUE TITLE
Backdropcms.org does not encrypt outgoing messages

### ISSUE THREAD

**Issue by alanmels** [backdrop-ops/backdropcms.org#608]
This one is kind of related to https://github.com/backdrop-ops/backdropcms.org/issues/607, but still a slightly different issue that deservers its own issue page.

Now, first let me quote from http://www.postfix.org/TLS_README.html#server_tls_auth:

> **Supporting AUTH over TLS only**
> Sending AUTH data over an unencrypted channel poses a security risk. When TLS layer encryption is required ("smtpd_tls_security_level = encrypt"), the Postfix SMTP server will announce and accept AUTH only after the TLS layer has been activated with STARTTLS. When TLS layer encryption is optional ("smtpd_tls_security_level = may"), it may however still be useful to only offer AUTH when TLS is active. To maintain compatibility with non-TLS clients, the default is to accept AUTH without encryption. In order to change this behavior, set "smtpd_tls_auth_only = yes".

And that's exactly how we used to set our Postfix mail server for `altagrade.com`: with `smtpd_tls_security_level = encrypt` and `smtpd_tls_auth_only = yes`, which effectively prevented all incoming e-mails from non-TLS senders (read as _badly configured or insecure servers_). Unfortunately, with the recommended settings above we never could receive anything from backdropcms.org, and because it's important for us not to miss any correspondence from Backdrop CMS project, we had to lower our security measures and start accepting non-TLS messages.

If you have a gmail account, then you can test the above problem by just sending anything to yourself from backdropcms.org (for example, request a new password on https://backdropcms.org/user/password) and you will see that even Google warns: 

> backdropcms.org did not encrypt this message

if you hover your mouse over the crossed out red lock sign as shown on the screenshot:

![IMG_2995](https://user-images.githubusercontent.com/56604405/68981152-8fe87b00-0813-11ea-8052-8bf553de3e07.jpg)

With the enormous quantity of spam transferred over Internet, more and more mail servers are enforcing stricter rules for incoming messages, and I really wouldn't want Backdropcms.org to fail delivering its correspondence to some users on such secured mail servers.

Also, any reputable project would take proper care of mail-related configuration settings. I am not sure if you, guys, need additional hands, but just in case please note AltaGrade will gladly help to get all the server-side issues, including mail-related configurations, to set properly.

**Comment by alanmels** [comment]
Could anyone run some tests to confirm this?

**Comment by larsdesigns** [comment]
I enforced TLS using the suggestions above. I also made some adjustments to the OpenDKIM.

**Comment by alanmels** [comment]
@larsdesigns, thank you for enforcing TLS, however testing it again shows the issue still persists, so something else should be done.

Gmail still shows red lock sign and when you click _to me_ drop-down and then proceed to clicking to _Learn more_ as shown on the screenshot:

![gmail](https://user-images.githubusercontent.com/56604405/73013722-130aec00-3dd6-11ea-8e7c-fcb117a6c3dd.jpg)

then it takes you to https://support.google.com/mail/answer/6330403?visit_id=637154016747712668-2642163949&p=tls&hl=en&rd=1. Quote of relevant part from that page:

> **What the encryption icons mean**
> When you're sending or receiving messages, you can see the level of encryption a message has. The color of the icon will change based on the level of encryption.
> 
> Red (no encryption) No encryption. Unencrypted mail which is not secure. Past messages sent to the recipient's domain are used to predict whether the message you're sending won't be reliably encrypted.
> 
> **I see the red lock icon**
> 
> If you're writing a message and see the red lock icon, consider removing these addresses or deleting the confidential information. To see which address is unencrypted, click View Details.
> 
> If you received a message with the red lock icon and the message contained particularly sensitive content, let the sender know and they can contact their email service provider.

**Comment by larsdesigns** [comment]
When I view the message, there is no icon in gmail. 

![Screen Shot 2020-01-23 at 12 53 49 PM](https://user-images.githubusercontent.com/976705/73018790-911fc080-3ddf-11ea-80b3-7f685b63b2b2.png)

When I view the source of the message, I see that there is encryption. 

Are your messages being forwarded by a host outside of google?

**Comment by larsdesigns** [comment]
@alanmels, Can you post the message header?

**Comment by alanmels** [comment]
> Are your messages being forwarded by a host outside of google?

No, I am testing this logging into actual Gmail account on https://mail.google.com

> @alanmels, Can you post the message header?

Sure, here it is:


```
Delivered-To: some-email-account@gmail.com
Received: by 2002:a17:90a:e010:0:0:0:0 with SMTP id u16csp1006958pjy;
        Thu, 23 Jan 2020 10:40:08 -0800 (PST)
X-Google-Smtp-Source: APXvYqwQhIbjB3zY4g+eMH2M+AWFlVyndCAkq4x6UqYVQLQPzmZnyWxebib9xWVKfUaGnwPdhHdD
X-Received: by 2002:a37:548:: with SMTP id 69mr17864990qkf.222.1579804808104;
        Thu, 23 Jan 2020 10:40:08 -0800 (PST)
ARC-Seal: i=1; a=rsa-sha256; t=1579804808; cv=none;
        d=google.com; s=arc-20160816;
        b=zazRVyYCgwEX94mrMu4DbanBVAAF9ekKqrQPUHwmJJNfIDkpRdgp/VssuRv7pSzQ+4
         i9WKj0wYF+6XMUtY+npJ/ihkHuIojYTQDj0XyCoQ8XoP+BS8q83LtpzWdAbDypF3j/6d
         +A5u6AVnHSFDNPlQz8MEVxhE3+NbxAi5Yym7NZRuWCBC5pKrUsC64X5JfSIf//WlVHEb
         TNL73f5jQq6NOCeQdGxmyHLqfljildeEFV0KqsZuSg10+vSLhAMDtRid0f9u9jTcRfQF
         vNCOKXi9QMUHQmhWzHsGZWsN2hOOB7t8QchD786+7qasAvt3qn/FDlWf7IsblSg6l4u0
         Rccw==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=google.com; s=arc-20160816;
        h=date:message-id:from:sender:content-transfer-encoding:mime-version
         :subject:to;
        bh=5tO5VQjoxORKQ1duogiXNwQBeMlgdAPFsFiewBpCSIw=;
        b=JBBgngW9ZiIpGmwCLwNPxCj7hCQXtkuwAFStSKznGl38gB37WY1WNEMWwCxf+RGdC8
         WIDepJxM+6jdq/+QSOXE5vW1ZoDmwotvjxNe5i3H9ZGSgkLcm8t+fx9Z3Rt2IP7pTU2N
         P9I+iAs+XZCqcYegM6G3mlP1hZOg9DBYJZsZOjwV1EBBrBH45VPkzatncLYSu7vBhxmL
         PafVC1PKYxiiYyw33J5AYijHUmqbt00/PT7FbQ91d7YqlN7agrdvwQ3/vNuZRlpuS+XB
         w5tKVmj35h6xLmYE46ppU0U/iiJmOOAO0UFGBNkTCVWMHlF7phbbb0wGAcfkXiPje9Dm
         W10A==
ARC-Authentication-Results: i=1; mx.google.com;
       spf=pass (google.com: domain of no-reply@backdropcms.org designates 2600:3c03::f03c:91ff:fec2:d257 as permitted sender) smtp.mailfrom=no-reply@backdropcms.org
Return-Path: <no-reply@backdropcms.org>
Received: from backdropcms.org (backdropcms.org. [2600:3c03::f03c:91ff:fec2:d257])
        by mx.google.com with ESMTP id x5si753757qtp.115.2020.01.23.10.40.07
        for <some-email-account@gmail.com>;
        Thu, 23 Jan 2020 10:40:08 -0800 (PST)
Received-SPF: pass (google.com: domain of no-reply@backdropcms.org designates 2600:3c03::f03c:91ff:fec2:d257 as permitted sender) client-ip=2600:3c03::f03c:91ff:fec2:d257;
Authentication-Results: mx.google.com;
       spf=pass (google.com: domain of no-reply@backdropcms.org designates 2600:3c03::f03c:91ff:fec2:d257 as permitted sender) smtp.mailfrom=no-reply@backdropcms.org
Received: by backdropcms.org (Postfix, from userid 33) id C345D5D775; Thu, 23 Jan 2020 18:40:07 +0000 (UTC)
To: some-email-account@gmail.com
Subject: Password reset information for SomeName at Backdrop CMS
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8; format=flowed; delsp=yes
Content-Transfer-Encoding: 8Bit
X-Mailer: Backdrop CMS
Sender: no-reply@backdropcms.org
From: no-reply@backdropcms.org
Message-Id: <20200123184007.C345D5D775@backdropcms.org>
Date: Thu, 23 Jan 2020 18:40:07 +0000 (UTC)
```

**Comment by alanmels** [comment]
@larsdesigns, the header information above was from my test in the morning. However, after posting my previous comment I had run one more test and this time it passed through without showing red-lock. So whatever change you've made since morning did the trick. Thank you very much and we may consider this as closed.

**Comment by alanmels** [comment]
Pardon me, I was too hasty. I've requested a password once again and it came with the red lock, unfortunately and the following header information:

```
Delivered-To: some-email-address@gmail.com
Received: by 2002:a17:90a:e010:0:0:0:0 with SMTP id u16csp1228994pjy;
        Thu, 23 Jan 2020 14:42:59 -0800 (PST)
X-Google-Smtp-Source: APXvYqwu0RlDEz9UYbjvwYgjT5qy2FkXKfYF/fnR4TAfzBdmC/dlUb9za3shDWne48wPvDlJ8oze
X-Received: by 2002:ac8:7cb0:: with SMTP id z16mr557866qtv.276.1579819379585;
        Thu, 23 Jan 2020 14:42:59 -0800 (PST)
ARC-Seal: i=1; a=rsa-sha256; t=1579819379; cv=none;
        d=google.com; s=arc-20160816;
        b=X5PzgauJxE6PXexPbO+CS+Zry2lU7WmhRb8FuzmQRWzdiK838lWRqC+wF6Lm5RHFut
         00+q/NBruJw7eNOTjw134lZ3ehqASM2WHAj5LhJhbaVlFOTVE1gaPlfUqI3ZvXcCAU1j
         gET01wCTbf1p5v79F+U6hCP3W0X7skYtcxXXo8CI0gnR5bmpxQNYs1UIY/lSXbvFtpyW
         cl6DMuU9UCP1LbVtLel4HCRt4PEzHZH4Dy92xovCEQJUJmORAAMgD47oJdUA9WibAkDd
         9xov9esSUZgbughxXQlZWmfCv5TbfsDYBiN0mD1Ud1M2GlryeUodpFRbQbI0Niv/q8rZ
         h1HQ==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=google.com; s=arc-20160816;
        h=date:message-id:from:sender:content-transfer-encoding:mime-version
         :subject:to;
        bh=1FNFTybfpar18o5qUCzUP830oPbmIEQN19VJMcXc7NQ=;
        b=a9Z6Jy9wCapUC5aobAA4w76cZO1tMbmNCbCGf2gnY6wGxMkw0KhXePRBHXBOrmlhmy
         w3Z9lUq9B47FWEiKfehAgtPakbsY0jcl0c+71dZKs1Y3CjKp3ETEug0m2G9Gq62oCbtU
         ZgtttRaZjaUZtFC3yMCuB8CoPD/Ze2jahnAokWoSMbbcrWd0TLkj17ptgl0J83Mn76TN
         0pIx7RCLp/nbM2aIpKteqjIexTQxrRlG5T0BA88CguWT79Rc8rLhRKUPIbseFkcWcNy2
         0JvBkzTXgtFw/SL+kTvlqYGr2TjCitzBm8mdsImKCgaZR0St0SAksmS1k/weWL70jBmW
         nYiQ==
ARC-Authentication-Results: i=1; mx.google.com;
       spf=pass (google.com: domain of no-reply@backdropcms.org designates 66.175.208.83 as permitted sender) smtp.mailfrom=no-reply@backdropcms.org
Return-Path: <no-reply@backdropcms.org>
Received: from backdropcms.org (backdropcms.org. [66.175.208.83])
        by mx.google.com with ESMTP id y62si2349236qtd.42.2020.01.23.14.42.59
        for <some-email-address@gmail.com>;
        Thu, 23 Jan 2020 14:42:59 -0800 (PST)
Received-SPF: pass (google.com: domain of no-reply@backdropcms.org designates 66.175.208.83 as permitted sender) client-ip=66.175.208.83;
Authentication-Results: mx.google.com;
       spf=pass (google.com: domain of no-reply@backdropcms.org designates 66.175.208.83 as permitted sender) smtp.mailfrom=no-reply@backdropcms.org
Received: by backdropcms.org (Postfix, from userid 33) id 5D9C45D775; Thu, 23 Jan 2020 22:42:59 +0000 (UTC)
To: some-email-address@gmail.com
Subject: Password reset information for SomeName at Backdrop CMS
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8; format=flowed; delsp=yes
Content-Transfer-Encoding: 8Bit
X-Mailer: Backdrop CMS
Sender: no-reply@backdropcms.org
From: no-reply@backdropcms.org
Message-Id: <20200123224259.5D9C45D775@backdropcms.org>
Date: Thu, 23 Jan 2020 22:42:59 +0000 (UTC)

SomeName,

A request to reset the password for your account has been made at Backdrop
CMS.

You may now reset your password by clicking this link or copying and pasting
it to your browser:

https://backdropcms.org/user/reset/2699/1579819379/0J_XjoLbLusSHsrMj7x0TWOFP8IfDXlUuwj65u0-xYA

This link can only be used once and will lead you to a page where you can set
your password. It expires after one day and nothing will happen if it is not
used.

```

So I am afraid I have to re-open this and request others to run some test using Gmail e-mail accounts.

**Comment by larsdesigns** [comment]
I think it will improve after we create the dkim dns record. 

Also, we are on track to use a third party smtp relay soon.

**Comment by alanmels** [comment]
This one also has been resolved by https://github.com/backdrop-ops/backdropcms.org/issues/607 Thanks a lot, @larsdesigns!
